experimental/devices/as7262/as7262.go

```diff
@@ -238,16 +254,44 @@ const (
 	G64x Gain = 0x30
 )
 
+const (
+	_GainG1x  = "1x"
+	_GainG4x  = "3.7x"
+	_GainG16x = "16x"
+	_GainG64x = "64x"
+)
+
+func (g Gain) String() string {
+	switch {
+	case g == 0:
+		return _GainG1x
+	case g == 16:
+		return _GainG4x
+	case g == 32:
+		return _GainG16x
+	case g == 48:
+		return _GainG64x
+	default:
+		return "bad gain value"
+	}
+}
+
 // Gain sets the gain of the sensor. There are four levels of gain 1x, 3.7x, 16x,
 // and 64x.
 func (d *Dev) Gain(gain Gain) error {
-	// TODO(NeuralSpaz): check that value is valid before writing. Currently
-	// a client could cast any int as Gain.
+	if gain != G1x && gain != G4x && gain != G16x && gain != G64x {
+		return errGainValue
+	}
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
-	d.ctx, d.cancel = context.WithCancel(context.Background())
-	defer d.cancel()
+	d.canceledMu.Lock()
```

Comment:
By the fact that d.mu is held, you know that Sense() is not running. Thus ines 288-294 are not needed.

Reply:
I think that is still needed in the case where Gain() is called after a Halt().